pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -400,6 +400,30 @@ func (p *Provider) GetHostErrorVMs(
 	return hostErrorVMs, nil
 }
 
+// GetVMSpecs returns a json list of VM specs
+func (p *Provider) GetVMSpecs(l *logger.Logger, vms vm.List) ([]map[string]interface{}, error) {
+	if p.GetProject() == "" {
+		return nil, errors.New("project name cannot be empty")
+	}
+	if vms == nil {
+		return nil, errors.New("vms cannot be nil")
+	}
+	// Extract the spec of all VMs.
+	var vmSpecs []map[string]interface{}
+	for _, vmInstance := range vms {
+		var vmSpec map[string]interface{}
+		vmFullResourceName := "projects/" + p.GetProject() + "/zones/" + vmInstance.Zone + "/instances/" + vmInstance.Name
+		args := []string{"compute", "instances", "describe", vmFullResourceName, "--format=json"}
+
+		if err := runJSONCommand(args, &vmSpec); err != nil {
+			l.Errorf("Error describing instance %s in zone %s: %v", vmInstance.Name, vmInstance.Zone, err)
+			return nil, err
```

Comment:
Nit: drop the `Errorf` call and just `return nil, errors.Wrap(err, "describing instance %s...")`

Reply:
And in that sense, this function might not need a logger instance.